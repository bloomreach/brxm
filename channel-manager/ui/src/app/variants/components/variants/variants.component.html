<!--
  Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  -->

<div class="mat-typography">
  <div class="variant-select">
    <mat-form-field appearance="outline">
      <mat-label class="qa-variant-select-label">
        {{ 'VARIANT' | translate }}

      </mat-label>
      <mat-select [formControl]="variantSelect"
                  class="qa-variant-select"
                  (selectionChange)="selectVariant($event.value)">
        <mat-option *ngFor="let variant of variants"
                    class="{{ 'qa-variant-select-option-' + variant.id }}"
                    [value]="variant">
          {{ variant.variantName }}
        </mat-option>
      </mat-select>
    </mat-form-field>

    <button mat-icon-button
            class="qa-variant-copy"
            (click)="addVariant()">
      <mat-icon svgIcon="copy"></mat-icon>
    </button>

    <button mat-icon-button
            class="qa-variant-delete">
      <mat-icon svgIcon="trash"></mat-icon>
    </button>
  </div>

  <div class="variant-expressions"
       *ngIf="currentVariant?.expressions.length > 0">
    <div class="variant-expressions-list">
      <div class="variant-expressions-list-label">
        Variant Expressions
      </div>

      <div class="variant-expressions-list-item"
           *ngFor="let expression of currentVariant?.expressions">
        <span>{{ expression.name }}</span>
        <button mat-icon-button (click)="removeExpression(expression)">
          <mat-icon svgIcon="close"></mat-icon>
        </button>
      </div>
    </div>

    <div class="variant-expressions-actions">
      <button mat-stroked-button>
        <mat-icon svgIcon="plus"></mat-icon>
        Add segment
      </button>
    </div>
  </div>
</div>
