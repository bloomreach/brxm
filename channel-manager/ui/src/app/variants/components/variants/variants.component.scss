/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~@bloomreach/frontend-theme/palettes';
@import 'variables';

:host {
  display: block;
  background: map-get($grey, 200);
  padding: $spacing-full;
}

::ng-deep .mat-form-field-wrapper {
  padding-bottom: 0;
}

.variant-select {
  align-items: center;
  display: grid;
  gap: $spacing-half;
  grid-template-columns: 1fr auto auto;
}

.variant-default-message {
  padding: $spacing-full 0;

  > span {
    color: map-get($text-color, dim);
    display: inherit;
    transform: scale(0.85);
    transform-origin: left top;
  }
}

.variant-expressions {
  padding-top: $spacing-full;

  &-list {
    $form-field-outline-color: rgba(0, 0, 0, 0.12);

    position: relative;
    border: 1px solid $form-field-outline-color;
    border-radius: 4px;
    margin-bottom: $spacing-full;
    padding: $spacing-half;

    &-label {
      color: map-get($text-color, dim);
      padding: 0 3px;
      position: absolute;
      background: map-get($grey, 200);
      font-size: 75%;
      top: -1em;
    }
  }

  &-list-item {
    display: grid;
    grid-template-columns: 1fr auto;
    align-items: center;
  }

  &-actions {
    display: flex;
    gap: $spacing-half;
  }
}
