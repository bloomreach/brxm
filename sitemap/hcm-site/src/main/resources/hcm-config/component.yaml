definitions:
  config:
    /hst:hst/hst:configurations/hst:default/hst:components/forge-sitemap-feed:
      jcr:primaryType: hst:component
      hst:componentclassname: org.onehippo.forge.sitemap.components.SitemapFeed
      hst:template: forge-sitemap-feed.ftl
