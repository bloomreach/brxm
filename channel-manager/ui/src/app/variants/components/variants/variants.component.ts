/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Component, Inject, OnInit } from '@angular/core';
import { switchMap } from 'rxjs/operators';

import { Ng1ComponentEditorService, NG1_COMPONENT_EDITOR_SERVICE } from '../../../services/ng1/component-editor.ng1.service';
import { VariantsService } from '../../services/variants.service';

@Component({
  selector: 'em-variants',
  templateUrl: './variants.component.html',
  styleUrls: ['./variants.component.scss'],
})
export class VariantsComponent implements OnInit {
  variants$ = this.variantsService
    .getVariantIds(this.componentEditorService.getComponentId())
    .pipe(
      switchMap(variantIds => this.variantsService.getVariants(variantIds)),
    );

  constructor(
    @Inject(NG1_COMPONENT_EDITOR_SERVICE) private readonly componentEditorService: Ng1ComponentEditorService,
    private readonly variantsService: VariantsService,
  ) { }

  ngOnInit(): void {
    this.variants$.subscribe(console.log);
  }

}
