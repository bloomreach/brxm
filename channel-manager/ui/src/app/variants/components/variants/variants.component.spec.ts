import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { VariantsComponent } from './variants.component';

describe('VariantsComponent', () => {
  let component: VariantsComponent;
  let fixture: ComponentFixture<VariantsComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ VariantsComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(VariantsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
