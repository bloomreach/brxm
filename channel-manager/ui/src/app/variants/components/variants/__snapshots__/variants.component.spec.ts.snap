// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantsComponent collapsing of variant expressions should hide expressions for non-default variant if toggle is clicked 1`] = `
<div
  id="root8"
>
  <div
    class="mat-typography"
  >
    <div
      class="variant-select"
    >
      <mat-form-field
        appearance="outline"
        class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label ng-untouched ng-pristine ng-valid mat-form-field-should-float"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-outline"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            <div
              class="mat-form-field-outline mat-form-field-outline-thick"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-autocomplete="none"
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="true"
                aria-invalid="false"
                aria-required="false"
                class="qa-variant-select mat-select ng-untouched ng-pristine ng-valid"
                role="combobox"
                tabindex="0"
              >
                <div
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    
                    
                    <span
                      class="mat-select-value-text"
                    >
                      
                      <span
                        class="mat-select-min-line"
                      >
                        Dutch
                      </span>
                      
                    </span>
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  class="mat-form-field-label"
                >
                  
                  
                  <mat-label
                    class="qa-variant-select-label"
                  >
                     VARIANT 
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <button
        aria-describedby="cdk-describedby-message-33"
        cdk-describedby-host=""
        class="qa-variant-copy mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="copy"
        >
          copy
        </mat-icon>
      </button>
      <button
        aria-describedby="cdk-describedby-message-34"
        cdk-describedby-host=""
        class="qa-variant-delete mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="trash"
        >
          trash
        </mat-icon>
      </button>
    </div>
    
    
    
    <div
      class="qa-variant-expressions-toggle variant-expressions-toggle"
    >
      <button
        class="qa-variant-expressions-toggle-button"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="chevron-right"
        >
          chevron-right
        </mat-icon>
      </button>
    </div>
  </div>
</div>
`;

exports[`VariantsComponent collapsing of variant expressions should not show expressions for default variant 1`] = `
<div
  id="root6"
>
  <div
    class="mat-typography"
  >
    <div
      class="variant-select"
    >
      <mat-form-field
        appearance="outline"
        class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-outline"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            <div
              class="mat-form-field-outline mat-form-field-outline-thick"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-autocomplete="none"
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="true"
                aria-invalid="false"
                aria-required="false"
                class="qa-variant-select mat-select mat-select-empty ng-untouched ng-pristine ng-valid"
                role="combobox"
                tabindex="0"
              >
                <div
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    
                    <span
                      class="mat-select-placeholder mat-select-min-line"
                    >
                      
                    </span>
                    
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  class="mat-form-field-label mat-empty mat-form-field-empty"
                >
                  
                  
                  <mat-label
                    class="qa-variant-select-label"
                  >
                     VARIANT 
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <button
        aria-describedby="cdk-describedby-message-24"
        cdk-describedby-host=""
        class="qa-variant-copy mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="copy"
        >
          copy
        </mat-icon>
      </button>
      <button
        aria-describedby="cdk-describedby-message-25"
        cdk-describedby-host=""
        class="qa-variant-delete mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="trash"
        >
          trash
        </mat-icon>
      </button>
    </div>
    
    
    <div
      class="qa-variant-expressions variant-expressions"
    >
      <div
        class="variant-expressions-list"
      >
        <div
          class="variant-expressions-list-label"
        >
           VARIANT_EXPRESSIONS 
        </div>
        
        
      </div>
      <div
        class="variant-expressions-actions"
      >
        <button
          aria-describedby="cdk-describedby-message-26"
          cdk-describedby-host=""
          class="qa-variant-expressions-add-segment mat-tooltip-trigger"
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           SEGMENT 
        </button>
        <button
          aria-describedby="cdk-describedby-message-27"
          cdk-describedby-host=""
          class="qa-variant-expressions-add-characteristic mat-tooltip-trigger"
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           CHARACTERISTIC 
        </button>
      </div>
    </div>
    
    <div
      class="qa-variant-expressions-toggle variant-expressions-toggle open"
    >
      <button
        class="qa-variant-expressions-toggle-button"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="chevron-right"
        >
          chevron-right
        </mat-icon>
      </button>
    </div>
  </div>
</div>
`;

exports[`VariantsComponent collapsing of variant expressions should show expressions for non-default variant 1`] = `
<div
  id="root7"
>
  <div
    class="mat-typography"
  >
    <div
      class="variant-select"
    >
      <mat-form-field
        appearance="outline"
        class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-outline"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            <div
              class="mat-form-field-outline mat-form-field-outline-thick"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-autocomplete="none"
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="true"
                aria-invalid="false"
                aria-required="false"
                class="qa-variant-select mat-select mat-select-empty ng-untouched ng-pristine ng-valid"
                role="combobox"
                tabindex="0"
              >
                <div
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    
                    <span
                      class="mat-select-placeholder mat-select-min-line"
                    >
                      
                    </span>
                    
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  class="mat-form-field-label mat-empty mat-form-field-empty"
                >
                  
                  
                  <mat-label
                    class="qa-variant-select-label"
                  >
                     VARIANT 
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <button
        aria-describedby="cdk-describedby-message-28"
        cdk-describedby-host=""
        class="qa-variant-copy mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="copy"
        >
          copy
        </mat-icon>
      </button>
      <button
        aria-describedby="cdk-describedby-message-29"
        cdk-describedby-host=""
        class="qa-variant-delete mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="trash"
        >
          trash
        </mat-icon>
      </button>
    </div>
    
    
    <div
      class="qa-variant-expressions variant-expressions"
    >
      <div
        class="variant-expressions-list"
      >
        <div
          class="variant-expressions-list-label"
        >
           VARIANT_EXPRESSIONS 
        </div>
        
        
        <div
          class="variant-expressions-list-item"
        >
          <span>
             EXPRESSION_SEGMENT EXPRESSION_AND 
          </span>
          <button
            aria-describedby="cdk-describedby-message-32"
            cdk-describedby-host=""
            class="qa-variant-expressions-delete mat-tooltip-trigger"
            mat-icon-button=""
          >
            <mat-icon
              svgicon="close"
            >
              close
            </mat-icon>
          </button>
        </div>
        <div
          class="variant-expressions-list-item"
        >
          <span>
             EXPRESSION_TARGET_GROUP_COUNTRY EXPRESSION_AND 
          </span>
          <button
            aria-describedby="cdk-describedby-message-32"
            cdk-describedby-host=""
            class="qa-variant-expressions-delete mat-tooltip-trigger"
            mat-icon-button=""
          >
            <mat-icon
              svgicon="close"
            >
              close
            </mat-icon>
          </button>
        </div>
        <div
          class="variant-expressions-list-item"
        >
          <span>
             EXPRESSION_TARGET_GROUP_CONTINENT  
          </span>
          <button
            aria-describedby="cdk-describedby-message-32"
            cdk-describedby-host=""
            class="qa-variant-expressions-delete mat-tooltip-trigger"
            mat-icon-button=""
          >
            <mat-icon
              svgicon="close"
            >
              close
            </mat-icon>
          </button>
        </div>
      </div>
      <div
        class="variant-expressions-actions"
      >
        <button
          aria-describedby="cdk-describedby-message-30"
          cdk-describedby-host=""
          class="qa-variant-expressions-add-segment mat-tooltip-trigger"
          disabled=""
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           SEGMENT 
        </button>
        <button
          aria-describedby="cdk-describedby-message-31"
          cdk-describedby-host=""
          class="qa-variant-expressions-add-characteristic mat-tooltip-trigger"
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           CHARACTERISTIC 
        </button>
      </div>
    </div>
    
    <div
      class="qa-variant-expressions-toggle variant-expressions-toggle open"
    >
      <button
        class="qa-variant-expressions-toggle-button"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="chevron-right"
        >
          chevron-right
        </mat-icon>
      </button>
    </div>
  </div>
</div>
`;

exports[`VariantsComponent collapsing of variant expressions should show expressions for non-default variant if toggle is clicked again 1`] = `
<div
  id="root9"
>
  <div
    class="mat-typography"
  >
    <div
      class="variant-select"
    >
      <mat-form-field
        appearance="outline"
        class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label ng-untouched ng-pristine ng-valid mat-form-field-should-float"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-outline"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            <div
              class="mat-form-field-outline mat-form-field-outline-thick"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-autocomplete="none"
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="true"
                aria-invalid="false"
                aria-required="false"
                class="qa-variant-select mat-select ng-untouched ng-pristine ng-valid"
                role="combobox"
                tabindex="0"
              >
                <div
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    
                    
                    <span
                      class="mat-select-value-text"
                    >
                      
                      <span
                        class="mat-select-min-line"
                      >
                        Dutch
                      </span>
                      
                    </span>
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  class="mat-form-field-label"
                >
                  
                  
                  <mat-label
                    class="qa-variant-select-label"
                  >
                     VARIANT 
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <button
        aria-describedby="cdk-describedby-message-38"
        cdk-describedby-host=""
        class="qa-variant-copy mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="copy"
        >
          copy
        </mat-icon>
      </button>
      <button
        aria-describedby="cdk-describedby-message-39"
        cdk-describedby-host=""
        class="qa-variant-delete mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="trash"
        >
          trash
        </mat-icon>
      </button>
    </div>
    
    
    <div
      class="qa-variant-expressions variant-expressions"
    >
      <div
        class="variant-expressions-list"
      >
        <div
          class="variant-expressions-list-label"
        >
           VARIANT_EXPRESSIONS 
        </div>
        
        
        <div
          class="variant-expressions-list-item"
        >
          <span>
             EXPRESSION_SEGMENT EXPRESSION_AND 
          </span>
          <button
            class="qa-variant-expressions-delete mat-tooltip-trigger"
            mat-icon-button=""
          >
            <mat-icon
              svgicon="close"
            >
              close
            </mat-icon>
          </button>
        </div>
        <div
          class="variant-expressions-list-item"
        >
          <span>
             EXPRESSION_TARGET_GROUP_COUNTRY EXPRESSION_AND 
          </span>
          <button
            class="qa-variant-expressions-delete mat-tooltip-trigger"
            mat-icon-button=""
          >
            <mat-icon
              svgicon="close"
            >
              close
            </mat-icon>
          </button>
        </div>
        <div
          class="variant-expressions-list-item"
        >
          <span>
             EXPRESSION_TARGET_GROUP_CONTINENT  
          </span>
          <button
            class="qa-variant-expressions-delete mat-tooltip-trigger"
            mat-icon-button=""
          >
            <mat-icon
              svgicon="close"
            >
              close
            </mat-icon>
          </button>
        </div>
      </div>
      <div
        class="variant-expressions-actions"
      >
        <button
          class="qa-variant-expressions-add-segment mat-tooltip-trigger"
          disabled=""
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           SEGMENT 
        </button>
        <button
          class="qa-variant-expressions-add-characteristic mat-tooltip-trigger"
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           CHARACTERISTIC 
        </button>
      </div>
    </div>
    
    <div
      class="qa-variant-expressions-toggle variant-expressions-toggle open"
    >
      <button
        class="qa-variant-expressions-toggle-button"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="chevron-right"
        >
          chevron-right
        </mat-icon>
      </button>
    </div>
  </div>
</div>
`;

exports[`VariantsComponent should create 1`] = `
<div
  id="root0"
>
  <div
    class="mat-typography"
  >
    <div
      class="variant-select"
    >
      <mat-form-field
        appearance="outline"
        class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-outline"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            <div
              class="mat-form-field-outline mat-form-field-outline-thick"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-autocomplete="none"
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="true"
                aria-invalid="false"
                aria-required="false"
                class="qa-variant-select mat-select mat-select-empty ng-untouched ng-pristine ng-valid"
                role="combobox"
                tabindex="0"
              >
                <div
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    
                    <span
                      class="mat-select-placeholder mat-select-min-line"
                    >
                      
                    </span>
                    
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  class="mat-form-field-label mat-empty mat-form-field-empty"
                >
                  
                  
                  <mat-label
                    class="qa-variant-select-label"
                  >
                     VARIANT 
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <button
        aria-describedby="cdk-describedby-message-0"
        cdk-describedby-host=""
        class="qa-variant-copy mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="copy"
        >
          copy
        </mat-icon>
      </button>
      <button
        aria-describedby="cdk-describedby-message-1"
        cdk-describedby-host=""
        class="qa-variant-delete mat-tooltip-trigger"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="trash"
        >
          trash
        </mat-icon>
      </button>
    </div>
    
    
    <div
      class="qa-variant-expressions variant-expressions"
    >
      <div
        class="variant-expressions-list"
      >
        <div
          class="variant-expressions-list-label"
        >
           VARIANT_EXPRESSIONS 
        </div>
        
        
      </div>
      <div
        class="variant-expressions-actions"
      >
        <button
          aria-describedby="cdk-describedby-message-2"
          cdk-describedby-host=""
          class="qa-variant-expressions-add-segment mat-tooltip-trigger"
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           SEGMENT 
        </button>
        <button
          aria-describedby="cdk-describedby-message-3"
          cdk-describedby-host=""
          class="qa-variant-expressions-add-characteristic mat-tooltip-trigger"
          mat-stroked-button=""
        >
          <mat-icon
            svgicon="plus"
          >
            plus
          </mat-icon>
           CHARACTERISTIC 
        </button>
      </div>
    </div>
    
    <div
      class="qa-variant-expressions-toggle variant-expressions-toggle open"
    >
      <button
        class="qa-variant-expressions-toggle-button"
        mat-icon-button=""
      >
        <mat-icon
          svgicon="chevron-right"
        >
          chevron-right
        </mat-icon>
      </button>
    </div>
  </div>
</div>
`;

exports[`VariantsComponent when the component editor is readonly should disable the action buttons 1`] = `
<div
  id="root5"
>
  <div
    class="mat-typography"
  >
    <div
      class="variant-select"
    >
      <mat-form-field
        appearance="outline"
        class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-outline"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            <div
              class="mat-form-field-outline mat-form-field-outline-thick"
            >
              <div
                class="mat-form-field-outline-start"
              />
              <div
                class="mat-form-field-outline-gap"
              />
              <div
                class="mat-form-field-outline-end"
              />
            </div>
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-autocomplete="none"
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="true"
                aria-invalid="false"
                aria-required="false"
                class="qa-variant-select mat-select mat-select-empty ng-untouched ng-pristine ng-valid"
                role="combobox"
                tabindex="0"
              >
                <div
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    
                    <span
                      class="mat-select-placeholder mat-select-min-line"
                    >
                      
                    </span>
                    
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  class="mat-form-field-label mat-empty mat-form-field-empty"
                >
                  
                  
                  <mat-label
                    class="qa-variant-select-label"
                  >
                     VARIANT 
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <button
        aria-describedby="cdk-describedby-message-20"
        cdk-describedby-host=""
        class="qa-variant-copy mat-tooltip-trigger"
        disabled=""
        mat-icon-button=""
      >
        <mat-icon
          svgicon="copy"
        >
          copy
        </mat-icon>
      </button>
      <button
        aria-describedby="cdk-describedby-message-21"
        cdk-describedby-host=""
        class="qa-variant-delete mat-tooltip-trigger"
        disabled=""
        mat-icon-button=""
      >
        <mat-icon
          svgicon="trash"
        >
          trash
        </mat-icon>
      </button>
    </div>
    
    <div
      class="qa-variant-default-message variant-default-message"
    >
      <span>
        VARIANT_EXPRESSIONS_DEFAULT
      </span>
    </div>
    
    
  </div>
</div>
`;
